euphony/src/main/java/co/euphony/tx/EuTxManager.kt

```diff
@@ -0,0 +1,185 @@
+
+    private fun playWithNativeEngine(count: Int) {
+        txCore.setCountToneOn(true, count)
+        txCore.tx_start()
+    }
+
+    private fun playWithAndroidEngine(count: Int) {
+        var loopCount = count
+        val outStream = txCore.genWaveSource
+        val minBufferSizeBytes = AudioTrack.getMinBufferSize(
+            Constants.SAMPLERATE,
+            AudioFormat.CHANNEL_OUT_MONO,
+            AudioFormat.ENCODING_PCM_FLOAT
+        )
+        val bufferSize = outStream.size * minBufferSizeBytes
+        audioTrack = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            AudioTrack.Builder()
+                .setAudioFormat(
+                    AudioFormat.Builder()
+                        .setChannelMask(AudioFormat.CHANNEL_OUT_MONO)
+                        .setSampleRate(Constants.SAMPLERATE)
+                        .setEncoding(AudioFormat.ENCODING_PCM_FLOAT)
+                        .build()
+                )
+                .setAudioAttributes(
+                    AudioAttributes.Builder()
+                        .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
+                        .build()
+                )
+                .setBufferSizeInBytes(bufferSize)
+                .setTransferMode(AudioTrack.MODE_STATIC)
+                .build()
+        } else {
+            AudioTrack(
+                AudioManager.STREAM_MUSIC,
+                Constants.SAMPLERATE,
+                AudioFormat.CHANNEL_OUT_MONO,
+                AudioFormat.ENCODING_PCM_FLOAT,
+                bufferSize,
+                AudioTrack.MODE_STATIC
+            )
+        }
+
+        /* A value of -1 means infinite looping, and 0 disables looping. */
+        if (loopCount <= 0) loopCount = -1
+        val result = audioTrack?.setLoopPoints(0, outStream.size, loopCount)
+        if (result != AudioTrack.SUCCESS) {
+            Log.i("PROCESS", "failed to loop points : $result")
+        }
+        audioTrack?.let {
+            try {
+                it.write(outStream, 0, outStream.size, AudioTrack.WRITE_NON_BLOCKING)
+                it.play()
+            } catch (e: IllegalStateException) {
+                Log.i("PROCESS", e.message ?: "")
+            }
+        }
+    }
+
+    fun stop() {
+        if (modeType == ModeType.DEFAULT && playerEngineType == PlayerEngine.ANDROID_DEFAULT_ENGINE) {
+            audioTrack?.pause()
+        } else {
+            txCore.tx_stop()
+        }
+    }
+
+    /*
+	 * @deprecated Replaced by {@link #setModeType(String)}, deprecated for naming & dynamic option.
+	 */
+    @Deprecated("", ReplaceWith(
+        "modeType = type",
+        "co.euphony.util.EuOption.ModeType"
+        )
+    )
+    fun setMode(type: ModeType?) {
+        modeType = type
+    }
+
+    /*
+	 * @deprecated Replaced by {@link #setCode(String)}, deprecated for naming & dynamic option.
+	 */
+    @Deprecated("", ReplaceWith(
+        "code = data",
+        "co.euphony.common.EuNativeConnector"
```

Comment:
I think you should use `setModeType(type)` and `setCode(data)` instead of `modeType = type` and `code = data`
The first argument should be a code piece that you want to replace. [(A post about this)](https://dev.to/mreichelt/the-hidden-kotlin-gem-you-didn-t-think-you-ll-love-deprecations-with-replacewith-3blo)

Reply:
Thank you for your reference link! I could understand the reference. I modified it and committed it! 👍🏻